From 6cdc3215d1691bcd2a407a30e88efa0cec179eb2 Mon Sep 17 00:00:00 2001
From: Marco Nelissen <marco.nelissen@gmail.com>
Date: Sat, 20 Aug 2022 23:03:09 -0700
Subject: [PATCH] arm64: dts: rockchip: enable Rock Pi 4C+ OTG

Rock Pi 4C+ is advertised as having one USB3 OTG port, so make it so.

Signed-off-by: Marco Nelissen <marco.nelissen@gmail.com>
---
 arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c-plus.dts | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c-plus.dts b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c-plus.dts
index 0107e85a23fdc..edb6eda277b71 100644
--- a/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c-plus.dts
+++ b/arch/arm64/boot/dts/rockchip/rk3399-rock-pi-4c-plus.dts
@@ -522,8 +522,7 @@
 };
 
 &tcphy0 {
-	extcon = <&virtual_pd>;
-	status = "disabled";
+	status = "okay";
 };
 
 &tcphy1 {
@@ -587,7 +586,8 @@
 
 &usbdrd_dwc3_0 {
 	status = "okay";
-	dr_mode = "host";
+	extcon = <&u2phy0>;
+	dr_mode = "otg";
 };
 
 &usbdrd3_1 {
